frontend/src/pages/SignUp/index.jsx

```diff
@@ -78,6 +53,46 @@ const SignUp = () => {
   const infoUpdateFailMessage = selector((state) => state.auth.infoUpdateFailMessage);
   const infoUpdateLoading = selector((state) => state.auth.infoUpdateLoading);
 
+  useEffect(() => {
+    console.log('stepchange');
+    if(formStep===0){
+      console.log('0');
+    } else if(formStep===1){
+      console.log('1');
+    } else if(formStep===2){
+      console.log('2');
+      dispatch(recommendations('tag', setTags));
+      dispatch(recommendations('uni', setUniversities));
+      dispatch(recommendations('dep', setDepartments));
+    }
+
+  },[formStep]);
+
+  useEffect(() => {
+    setInterestChoices([]);
+    setUniversityChoices([]);
+    setDepartmentChoices([]);
+    
+    var ic = [];
+    var uc = [];
+    var dc = [];
+    
+    for (let i = 0; i < tags.length; i++) {
+      ic.push(<Option key={tags[i]}>{tags[i]}</Option>);
+    }
+    for (let i = 0; i < universities.length; i++) {
+      uc.push(<Option key={universities[i]}>{universities[i]}</Option>);
+    }
+    for (let i = 0; i < departments.length; i++) {
+      dc.push(<Option key={departments[i]}>{departments[i]}</Option>);
+    }
+    setInterestChoices(ic);
+    setUniversityChoices(uc);
+    setDepartmentChoices(dc);
+  },[tags, universities, departments]);
```

Comment:
Didn't really understand your purpose here

Reply:
the multiselect antd component takes a list of `<Option />` objects, while, the endpoints return mere lists. I take those lists from the api returns and create `<Option />` object lists.

Reply:
We have edited this code to avoid unnecessary overhead on api call results. Resolving conversation.